tracker_blackbox_test.go

```diff
@@ -0,0 +1,207 @@
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getExpiredAuthHeader(t, privatekey),
+		}, {
+			method:             "PUT",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getMalformedAuthHeader(t, privatekey),
+		}, {
+			method:             "PUT",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getValidAuthHeader(t, differentPrivatekey),
+		}, {
+			method:             "PUT",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           "",
+		},
+		// Delete tracker API with different parameters
+		{
+			method:             "DELETE",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getExpiredAuthHeader(t, privatekey),
+		}, {
+			method:             "DELETE",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getMalformedAuthHeader(t, privatekey),
+		}, {
+			method:             "DELETE",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           getValidAuthHeader(t, differentPrivatekey),
+		}, {
+			method:             "DELETE",
+			url:                "/api/trackers/12345",
+			expectedStatusCode: 401,
+			expectedErrorCode:  "jwt_security_error",
+			payload:            createTrackerPayload,
+			jwtToken:           "",
+		},
+		// Try fetching a random tracker
+		// We do not have security on GET hence this should return 404 not found
+		{
+			method:             "GET",
+			url:                "/api/trackers/088481764871",
+			expectedStatusCode: 404,
+			expectedErrorCode:  "not_found",
+			payload:            nil,
+			jwtToken:           "",
+		},
+	}
+}
+
+// This test case will check authorized access to Create/Update/Delete APIs
+func TestUnauthorizeTrackerCUD(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	// This will be modified after merge PR for "Viper Environment configurations"
+	publickey, err := jwt.ParseRSAPublicKeyFromPEM((configuration.GetTokenPublicKey()))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+	tokenTests := getTrackerTestData(t)
+
+	scheduler := remoteworkitem.NewScheduler(DB)
+	for _, testObject := range tokenTests {
```

Comment:
For each iteration in the loop, could you print some information about what is the test case that is being executed so that in case of failure it might be easy to drill down?